packages/remix-dev/compiler/server/write.ts

```diff
@@ -14,7 +14,7 @@ export async function write(
     if ([".js", ".cjs", ".mjs"].some((ext) => file.path.endsWith(ext))) {
       // fix sourceMappingURL to be relative to current path instead of /build
       let filename = file.path.substring(file.path.lastIndexOf(path.sep) + 1);
-      let escapedFilename = filename.replace(/\./g, "\\.");
+      let escapedFilename = filename.replace(/([.[\]])/g, "\\$1");
```

Comment:
I like this regex better than the 3-pass replace in https://github.com/remix-run/remix/pull/5809, but do we need to also escape the `[` in the regex, as suggested in https://github.com/remix-run/remix/issues/3768#issuecomment-1186553615?

```suggestion
      let escapedFilename = filename.replace(/([.\[\]])/g, "\\$1");
```

Reply:
No, it's inside the existing group, so it's not needed.